lib/active_model/serializer/array_serializer.rb

```diff
@@ -10,10 +10,9 @@ class ArraySerializer
       def initialize(resources, options = {})
         @root = options[:root]
         @object = resources
+        parent_serializer = options.fetch(:_parent_serializer, ActiveModel::Serializer)
         @serializers = resources.map do |resource|
-          serializer_class = options.fetch(:serializer) do
-            ActiveModel::Serializer.serializer_for(resource)
-          end
+          serializer_class = options.fetch(:serializer) { parent_serializer.serializer_for(resource) }
 
```

Comment:
This is part of the dough of this PR (the whole `ArraySerializer` situation adds an intermediate object, so we need to keep track of the parent serializer).